### DISCUSSION TITLE
Disable boundary condition after time step

### DISCUSSION THREAD

**Discussion by mfitzka** [idaholab/moose#23579] [Q&A General] [answered]
Hello,

Is there a way to disable a boundary condition after a certain time frame?

I would like to have the temperature to be (for example) 100 °C at the beginning at one boundary and after a certain time period it changes back to room temperature. and you can see the temperature changing over time in the material. Similar to bringing the material in the mesh in contact with a hot external source but not in for the whole time as can be done with the standard BCs.

Or are there other ways to implement this, if this is not possible?

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can use the Control interface to turn on or off objects
https://mooseframework.inl.gov/syntax/Controls/index.html

You can then switch from a heat flux (NeumannBC) to a fixed temperature (DIrichlet) at a certain time. 
If the BCs are off the same type through time, with simply different values, then you just need to define the value as a Function, like FunctionDirichletBC. Functions can have time dependence

Guillaume

**Reply by mfitzka** [reply]
That should work perfectly for my problem. Thank you Guillaume!
